Duplicati/Library/Backend/OneDrive/MicrosoftGraphBackend.cs

```diff
@@ -269,21 +271,17 @@ public void CreateFolder()
 
         public IEnumerable<IFileEntry> List()
         {
-            try
-            {
-                return this.Enumerate<DriveItem>(string.Format("{0}/root:{1}:/children", this.DrivePrefix, this.RootPath))
-                    .Where(item => item.IsFile && !item.IsDeleted) // Exclude non-files and deleted items (not sure if they show up in this listing, but make sure anyway)
-                    .Select(item =>
-                        new FileEntry(
-                            item.Name,
-                            item.Size ?? 0, // Files should always have a size, but folders don't need it
-                            item.FileSystemInfo?.LastAccessedDateTime?.UtcDateTime ?? new DateTime(),
-                            item.FileSystemInfo?.LastModifiedDateTime?.UtcDateTime ?? item.LastModifiedDateTime?.UtcDateTime ?? new DateTime()));
-            }
-            catch (DriveItemNotFoundException ex)

```

Comment:
Can you briefly explain why this catch vanished?

Reply:
This catch didn't vanish, but was moved into the Enumerate method at the bottom of this file (this is the only place Enumerate is called).

The reason for this is mainly to make sure that the lazy evaluation of the select (and of any other requests) here don't cause the exception mapping from the internal one to Duplicati's FolderNotFound one to be missed. This also moves the exception handling closer to the source.

Reply:
I understand, but I don't see how the Enumerate method is connected to the List method. Can you explain?

Reply:
List now calls `foreach (DriveItem item in this.Enumerate<DriveItem>(...))` to iterate rather than returning `this.Enumerate<DriveItem>(...).Where(...).Select(...)`

Reply:
I meant the enumerate function below in the code, where the driveitemnotfound exception is caught and you referred to earlier. It involves a GraphCollection.